image/parse.go

```diff
@@ -161,13 +182,36 @@ func ParseImage(imgData []byte) (Image, error) {
 	}
 	offset += size
 
+	var protTrailer *ImageTrailer
+	var protTlvs []ImageTlv
+	if hdr.ProtSz > 0 {
+		pt, size, err := parseRawTrailer(imgData, offset)
+		if err != nil {
```

Comment:
I would suggest extending this to `err != nil && size != hdr.ProtSz`

Reply:
`parseRawTrailer()` just seems to return `IMAGE_TRAILER_SIZE` (4).  As far as I can tell, we have to trust `hdr.ProtSz` and just parse away!

Reply:
Hmmm, well as long as the bootloader check it, should not be a problem anyway! :-P